src/comm/SerialLink.cc

```diff
@@ -475,11 +411,17 @@ bool SerialLink::hardwareConnect(QString &type)
     QObject::connect(m_port, SIGNAL(aboutToClose()), this, SLOT(_rerouteDisconnected()));
     QObject::connect(m_port, SIGNAL(error(QSerialPort::SerialPortError)), this, SLOT(linkError(QSerialPort::SerialPortError)));
 
-    checkIfCDC();
-
     //    port->setCommTimeouts(QSerialPort::CtScheme_NonBlockingRead);
 
-    if (!m_port->open(QIODevice::ReadWrite)) {
+    // After the bootloader times out, it still can take a second or so for the USB driver to come up and make
+    // the port available for open. So we retry a few times to wait for it.
+    for (int openRetries=0; openRetries<4; openRetries++) {
+        if (!m_port->open(QIODevice::ReadWrite)) {
+            qDebug() << "Port open failed, retrying";
+            QGC::SLEEP::msleep(500);
+        }
```

Comment:
I know this PR is closed but because it got me confused, I figured it wouldn't hurt to point these out.
This loop doesn't get out when the connection succeeds. It just keeps trying to connect again...